apps/emqx_rule_engine/src/emqx_rule_monitor.erl

```diff
@@ -0,0 +1,109 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_rule_monitor).
+
+-behavior(gen_server).
+
+-include("rule_engine.hrl").
+-include_lib("emqx/include/logger.hrl").
+-logger_header("[Rule Monitor]").
+
+-define(T_RETRY, 60000).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-export([ start_link/0
+        , ensure_resource_started/1
+        ]).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init([]) ->
+    {ok, #{pids => #{}}}.
+
+ensure_resource_started(ResId) ->
+    gen_server:call(?MODULE, {create_restart_handler, resource, ResId}).
```

Comment:
gen_server:call by default uses 5 seconds timeout.

Reply:
The default timeout is OK I think because this process is hard to get overloaded.

Reply:
the consequence of timeout is an error exception, is that OK ?

Reply:
A possible issue is if the process is suspended by the release_handler, the refresh_resource will timeout. If so I'll change this to gen_server:cast.